yy_parser.go

```diff
@@ -132,11 +136,18 @@ func (parser *Parser) SetParserConfig(config ParserConfig) {
 	parser.EnableWindowFunc(config.EnableWindowFunction)
 	parser.SetStrictDoubleTypeCheck(config.EnableStrictDoubleTypeCheck)
 	parser.lexer.skipPositionRecording = config.SkipPositionRecording
+	parser.decoder = config.Decoder
 }
 
 // Parse parses a query string to raw ast.StmtNode.
 // If charset or collation is "", default charset and collation will be used.
 func (parser *Parser) Parse(sql, charset, collation string) (stmt []ast.StmtNode, warns []error, err error) {
+	if parser.decoder != nil {
+		sql, _, err = transform.String(parser.decoder, sql)
```

Comment:
I assume we can not transform the whole sql like this, e.g. `select _utf8mb4 '汉字'`, because we have `_utf8mb4` before the string so that this string should be interpreted as utf8mb4 encoding.